### PR TITLE
Directory re-structure for Self-Deployer and Workflow-Agent

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

## Proposed changes

- moved self-deployer and workflow-agent to litmus-portal root

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: litmus-portal/Makefile [hunk 1]
```diff
@@ -68,4 +68,22 @@ backend-services-checks:
 		echo "go mod tidy checking failed!" && echo "$${tidyRes}" \
 		&& echo "Please ensure you are using $$($(GO) version) for formatting code." \
 		&& exit 1; \
-	fi
\ No newline at end of file
+    fi
```
Comment (rajdas98) [inline litmus-portal/Makefile:71]: script is not indent

Comment (gdsoumya) [reply litmus-portal/Makefile:71]: I don't think there's any problem in the script, I ran the make file it works. If there was an indentation error it wouldn't work. Locally I can run the makefile : make backend-services-checks

```
------------------
--> checking code style [backend]
------------------
------------------
--> Check litmus-portal backend [go mod tidy]
------------------
------------------
--> Check litmus-portal auth [go mod tidy]
------------------
------------------
--> Check litmus-portal self-deployer [go mod tidy]
------------------
------------------
--> Check litmus-portal workflow-agent [go mod tidy]
------------------
```

Comment (gdsoumya) [reply litmus-portal/Makefile:71]: Even in the CI all the checks ran correctly

Comment (rajdas98) [reply litmus-portal/Makefile:71]: Selection_029
Check the script, it is not indented properly. Edit it from the GitHub GUI.

Comment (gdsoumya) [reply litmus-portal/Makefile]: It is a github problem, on my system the indentation is correct. I have replace all the 4 spaced tabs with 8 spaces, it should solve the problem

Comment (rajdas98) [reply litmus-portal/Makefile]: Can you edit it from GitHub IDE?

**Review by rajdas98 — APPROVED** [review]
LGTM

**Pull request merged**